src/providers/go.rs

```diff
@@ -110,7 +110,7 @@ fn version_number_to_pkg(version: &str) -> Result<Option<String>> {
 
     match matched_version {
         Some((_, pkg)) => Ok(Some(pkg.to_string())),
-        None => bail!("Go version {} is not available", version),
+        None => Ok(None),
```

Comment:
Sorry if I'm missing something, but couldn't lots of the logic be cleaned up if this match statement always returned the version? Where it currently returns None should potentially return the default version.